lib/ZnapZend.pm

```diff
@@ -87,13 +90,40 @@ has zLog => sub {
             message => sub {
                 my ($log, $level, @lines) = @_;
                 syslog($logLevels{$level}, @lines);
+                if ($level eq 'warn' || $level eq 'err') {
+                  push @{$self->logBuffer}, @lines;
+                }
             }
         );
     };
 
     return $log;
 };
 
+my $createMailWorker = sub {
+    my $self = shift;
+
+    my $cb;
+    $cb = sub {
+        if (@{$self->logBuffer} && $self->{mailto} && $self->{mailprog}) {
+            open (my $fh, '|-', 
+              $self->{mailprog}, '-s', 'ZnapZend warning/error', $self->{mailto});
+            print $fh join "\n", @{$self->logBuffer};
+            close $fh;
+        }
+            
+        $self->logBuffer([]);
+    };
+
+    if ($self->runonce){
+        #run immediately
+        $cb->();
+    }
+    else{
+        Mojo::IOLoop->timer(10 => $cb);
+    }
```

Comment:
at the moment you will send mail every 10 seconds ... I think it would be cool if you would send email if either (more than 1 minute has passed since the last email or no more new data has become available for more than 10 seconds) that should reduce the number of mails in case there is high logging activity